pkg/trace/traceutil/azure_test.go

```diff
@@ -6,74 +6,122 @@
+	setEnvVars(t, mockAppServiceEnv)
+	websiteOS := runtime.GOOS
+	// Not in a function app
+	linux := GetAppServicesTags()
+	t.Setenv("FUNCTIONS_WORKER_RUNTIME", "node")
+	functionApp := GetAppServicesTags()
+
+	assert.Equal(t, "1234abcd", linux[aasInstanceID])
+	assert.Equal(t, "test-instance", linux[aasInstanceName])
+	assert.Equal(t, websiteOS, linux[aasOperatingSystem])
+	assert.Equal(t, "Node.js", linux[aasRuntime])
+	assert.Equal(t, "test-resource-group", linux[aasResourceGroup])
+	assert.Equal(t, "/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/test-resource-group/providers/microsoft.web/sites/site-name-test", linux[aasResourceID])
+	assert.Equal(t, "app", linux[aasSiteKind])
+	assert.Equal(t, "site-name-test", linux[aasSiteName])
+	assert.Equal(t, "app", linux[aasSiteType])
+	assert.Equal(t, "00000000-0000-0000-0000-000000000000", linux[aasSubscriptionID])
+	assert.Equal(t, "", linux[aasFunctionRuntime])
+	assert.Equal(t, "~4", functionApp[aasFunctionRuntime])
+
 }
 
 func TestGetEnvOrUnknown(t *testing.T) {
-	unknownEnvVar := getEnvOrUnknown("", mockGetEnvVar)
+	unknownEnvVar := getEnvOrUnknown("")
 	assert.Equal(t, "unknown", unknownEnvVar)
 }
 
 func TestGetWindowsRuntime(t *testing.T) {
-	java := getRuntime("windows", func(s string) string {
-		if s == "WEBSITE_STACK" {
-			return "JAVA"
-		}
-		return ""
-	})
-	node := getRuntime("windows", func(s string) string {
-		if s == "WEBSITE_NODE_DEFAULT_VERSION" {
-			return "~18"
-		}
-		return ""
-	})
-	dotnet := getRuntime("windows", func(s string) string { return "" })
+	// Java
+	t.Setenv("WEBSITE_STACK", "JAVA")
+	java := getRuntime("windows")
+
+	// Node
+	os.Unsetenv("WEBSITE_STACK")
+	t.Setenv("WEBSITE_NODE_DEFAULT_VERSION", "18")
+	node := getRuntime("windows")
+
+	// .NET
+	os.Unsetenv("WEBSITE_NODE_DEFAULT_VERSION")
+	dotnet := getRuntime("windows")
+
+	// Unset
+	t.Setenv("WEBSITE_STACK", "")
+	unknown := getRuntime("windows")
 
-	assert.Equal(t, ".NET", dotnet)
 	assert.Equal(t, "Java", java)
 	assert.Equal(t, "Node.js", node)
 	assert.Equal(t, ".NET", dotnet)
+	assert.Equal(t, ".NET", unknown)
 }
 
 func TestGetLinuxRuntime(t *testing.T) {
-	docker := getRuntime("linux", func(s string) string { return "DOCKER" })
-	compose := getRuntime("linux", func(s string) string {
-		if s == "WEBSITE_STACK" {
-			return ""
-		}
-		if s == "DOCKER_SERVER_VERSION" {
-			return "19.03.15+azure"
-		}
-		return ""
-	})
-	java := getRuntime("linux", func(s string) string { return "JAVA" })
-	tomcat := getRuntime("linux", func(s string) string { return "TOMCAT" })
-	node := getRuntime("linux", func(s string) string { return "NODE" })
-	python := getRuntime("linux", func(s string) string { return "PYTHON" })
-	dotnet := getRuntime("linux", func(s string) string { return "DOTNETCORE" })
-	php := getRuntime("linux", func(s string) string { return "PHP" })
-	unknown := getRuntime("linux", func(s string) string { return "" })
+	// Docker
```

Comment:
nit: This might read better as separate t.Run() instances? That way you don't need to keep setting and unsetting the env vars?

Reply:
Regarding the integration tests, we test this in actual applications prior to release. So once this is merged, we will need a new agent released before we can release a new version of the azure app service extension. Prior to uploading the extension package to Nuget, we test in our self-monitoring apps.